modules/rostests/kmtests/include/kmt_test.h

```diff
@@ -327,6 +398,9 @@ static VOID KmtAddToLogBuffer(PKMT_RESULTBUFFER Buffer, PCSTR String, SIZE_T Len
     } while (InterlockedCompareExchange(&Buffer->LogBufferLength, NewLength, OldLength) != OldLength);
 
     memcpy(&Buffer->LogBuffer[OldLength], String, Length);
+#if defined(KMT_KERNEL_MODE) && !defined(KMT_STANDALONE_DRIVER)
+    KmtFlushLogBuffer();
```

Comment:
Hmm. One problem we might get here is that this is not thread-safe: this function adds to the length before it actually writes valid log data, so the flush could print out uninitialized stuff. Perhaps we need a second Length variable for valid data?